contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -420,23 +463,37 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, IERC677Rece
       revert CannotRemoveWithPendingRequests();
     }
 
-    _cancelSubscriptionHelper(subscriptionId, to);
-  }
-
-  function _cancelSubscriptionHelper(uint64 subscriptionId, address to) private {
-    Subscription memory subscription = s_subscriptions[subscriptionId];
-    uint96 balance = subscription.balance;
-    // NOTE: loop iterations are bounded by config.maxConsumers
-    // If no consumers, does nothing.
-    for (uint256 i = 0; i < subscription.consumers.length; ++i) {
-      delete s_consumers[subscription.consumers[i]][subscriptionId];
+    (uint96 balance, uint64 completedRequests, ) = _cancelSubscriptionHelper(subscriptionId);
+
+    (
+      uint16 subscriptionDepositCompletedRequests,
+      uint72 subscriptionDepositJuels,
+      address owner
+    ) = _getSubscriptionDepositDetails();
+
+    IERC20 linkToken = IERC20(i_linkToken);
+
+    if (completedRequests >= subscriptionDepositCompletedRequests) {
```

Comment:
Potential simplification: if insufficient requests were made, take the deposit (or less) first. Then proceed normally.